images/rag/embedding.py

```diff
@@ -100,36 +98,47 @@ async def main():
             storage_context = StorageContext.from_defaults(vector_store=vector_store)
 
-    print("** Configured storage context")
-
-    service_context = ServiceContext.from_defaults(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP,embed_model=args.model, llm=None)
-    print("** Configured service_context")
-
-    documents = SimpleDirectoryReader(input_files=load_docs(args.folder)).load_data()
-    print("** Loading docs ")
-
-    index = VectorStoreIndex.from_documents(
-        documents, storage_context=storage_context, service_context=service_context, show_progress=True
-    )
-    index.set_index_id(PRODUCT_INDEX)
-    index.storage_context.persist(persist_dir=PERSIST_FOLDER)
-    print("*** Completed  embeddings ")
-
-    end_time = time.time()
-    execution_time_seconds = end_time - start_time
-
-    print(f"** Total execution time in seconds: {execution_time_seconds}")
 
-    # creating metadata folder
-    metadata = {}
-
-    metadata["execution-time"] = execution_time_seconds
+    print("** Configured storage context")        
+    service_context = ServiceContext.from_defaults(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP,embed_model=args.model, llm='local')
+    folders = args.folders.split(" ")
+    index_list = [] 
+    total_execution = 0 
+    total_docs = 0
+    for folder in folders:
+        if os.path.exists(folder):
+        
+            print(f"** building vector store for {folder}")        
+            print("** Configured service_context")        
+            
+            documents = SimpleDirectoryReader(input_files=load_docs(folder)).load_data()
+            print("** Loading docs ")
+            index = VectorStoreIndex.from_documents(
+                documents, storage_context=storage_context, service_context=service_context, show_progress=True
+            )
+            
+            folder_index = folder.split("/")[-1]
+            index.set_index_id(folder_index)
+            index.storage_context.persist(persist_dir=PERSIST_FOLDER)
+            
+            print("*** Completed  embeddings ")
+            end_time = time.time()
+            execution_time_seconds = end_time - start_time        
+            print(f"** Total execution time in seconds: {execution_time_seconds}")
+            
+            index_list.append(folder_index)
+            total_execution += execution_time_seconds
+            total_docs += len(documents)
+        
+        
+    # creating metadata file 
+    metadata = {} 
+    metadata["execution-time"] = total_execution
     metadata["llm"] = 'local'
-    metadata["embedding-model"] = args.model
-    metadata["index_id"] = PRODUCT_INDEX
-
+    metadata["embedding-model"] = args.model 
+    metadata["available-indexes"] = index_list
     metadata["vector-db"] = args.vector_type
-    metadata["total-embedded-files"] = len(documents)
+    metadata["total-embedded-files"] =total_docs
 
     json_metadata = json.dumps(metadata)
 
```

Comment:
does it make sense to include repo name + tag/commit/release? (IMHO the `folder` points to cloned repository, right? so might be possible to ask Git)

Reply:
i dont think those details will be usfull to embeddging consumers